.github/scripts/patch.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash -x -e -o nounset
+
+# .github/patchs/.versions.sh should define all the versions of the dependencies that we are going to patch
+# This is used so that we can properly clone the upstream repositories.
+# This file should define the following variables:
+# OTEL_JAVA_VERSION. Tag of the opentelemetry-java repository to use. E.g.: JAVA_OTEL_JAVA_VERSION=v1.21.0
+# OTEL_JAVA_INSTRUMENTATION_VERSION. Tag of the opentelemetry-java-instrumentation repository to use, e.g.: OTEL_JAVA_INSTRUMENTATION_VERSION=v1.21.0
+# OTEL_JAVA_CONTRIB_VERSION. Tag of the opentelemetry-java-contrib repository. E.g.: OTEL_JAVA_CONTRIB_VERSION=v1.21.0
+# This script will fail if a variable that is supposed to exist is referenced.
+
+if [[ ! -f .github/patchs/versions ]]; then
+  echo "No versions file found. Skipping patching"
+  exit 0
+fi
+
+source .github/patchs/versions
+
+git config --global user.email "github-actions@github.com"
+git config --global user.name "github-actions"
```

Comment:
Maybe set this to something that makes clear it is this workflow as the actor?

Reply:
done